Cabal-syntax/src/Distribution/Types/TestSuite.hs

```diff
@@ -26,7 +26,8 @@ import qualified Distribution.Types.BuildInfo.Lens as L
 data TestSuite = TestSuite {
         testName      :: UnqualComponentName,
         testInterface :: TestSuiteInterface,
-        testBuildInfo :: BuildInfo
+        testBuildInfo :: BuildInfo,
+        testCodeGenerators :: [String]
```

Comment:
A comment would be in order explaining this new field, maybe linking to this PR.

In particular I wonder, should this field be non-revisable on Hackage?
(By default, I suppose, "yes, cannot change in a revision".  But on the other hand, what is the harm of revising this?  It's just for the test suite.  Also, version of dependencies can be arbitrary revised, to the point of completely changing functionality... Why not test generaters...?)

Reply:
our principle for allowing revisions has always been "by default, negative, and allow revision only if there's demonstrated need"

Reply:
Thanks for the justification.